packages/medusa/src/services/region.ts

```diff
@@ -221,23 +245,25 @@ class RegionService extends BaseService {
       this.taxProviderRepository_
     )
 
+    const region = { ...regionData } as DeepPartial<Region>
+
     if (region.tax_rate) {
-      this.validateTaxRate_(region.tax_rate)
+      this.validateTaxRate(region.tax_rate)
     }
 
-    if (region.countries) {
+    if (regionData.countries) {
       region.countries = await Promise.all(
-        region.countries.map((countryCode) =>
-          this.validateCountry_(countryCode, id)
+        regionData.countries!.map((countryCode) =>
+          this.validateCountry(countryCode, id!)
         )
       ).catch((err) => {
         throw err
       })
     }
 
-    if (region.tax_provider_id) {
+    if ((regionData as UpdateRegionInput).tax_provider_id) {
       const tp = await tpRepository.findOne({
-        where: { id: region.tax_provider_id },
+        where: { id: (regionData as UpdateRegionInput).tax_provider_id },
```

Comment:
question: Why do we need to typecast here? 🤔

Reply:
we accept `tax_provider_id` on the region update endpoint but not on the create endpoint so region data will have tax provider id only if it is of `UpdateRegionInput` type